externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/KyuubiSparkUtil.scala

```diff
@@ -95,7 +95,5 @@ object KyuubiSparkUtil extends Logging {
     }
   }
 
-  // Given that we are on the Spark SQL engine side, the [[org.apache.spark.SPARK_VERSION]] can be
-  // represented as the runtime version of the Spark SQL engine.
-  lazy val SPARK_ENGINE_RUNTIME_VERSION = SemanticVersion(org.apache.spark.SPARK_VERSION)
+  lazy val sparkRuntimeVersion: SemanticVersion = SemanticVersion(SPARK_VERSION)
```

Comment:
Hi @cfmcgrady , I'm generalizing Spark's runtime semantic version into `sparkRuntimeVersion` as the same as in other engines and modules. Here I prefer the `import` way rather than the full name inline, which still loads Spark version when needed. WDYT ?

Reply:
> I'm generalizing Spark's runtime semantic version in sparkRuntimeVersion as the same as in other engines and modules.

As stated in the code comment, the [[org.apache.spark.SPARK_VERSION]] denotes the runtime version of the Spark SQL engine when the suite is executed on the Spark SQL engine side. However, when executing the suite on the client side, it is not feasible to represent the runtime version using [[org.apache.spark.SPARK_VERSION]], as shown in https://github.com/apache/kyuubi/pull/4381

> Here I prefer the import way rather than the full name inline, which still loads Spark version when needed.

I have no opinion on whether to use the `import` style, but I prefer uppercase for constants.

Reply:
Let me try to summarize your suggested changes:
1. preferred uppercase for constants, eg. `SPARK_ENGINE_RUNTIME_VERSION` as it is.
2. change to get Spark's runtime version via [JDBC query "SELECT version()" as in `SparkVersionUtil`](https://github.com/apache/kyuubi/blob/ecc3b4af6149ff49faff5f5360b67a6459d84856/kyuubi-common/src/test/scala/org/apache/kyuubi/util/SparkVersionUtil.scala#L30)
3. make this `SPARK_ENGINE_RUNTIME_VERSION` private and marked as visible for testing, as it's used in tests only.

Reply:
I reverted this to the original code here, as `lazy val SPARK_ENGINE_RUNTIME_VERSION = SemanticVersion(SPARK_VERSION)`.